### PR TITLE
Fix Output for isprime

### PR DESCRIPTION
**Pull request by shishir-11**
#### References to other Issues or PRs

Fixes Issue #26489

#### Brief description of what is fixed or changed
isprime raises a value error when using floating point values instead of giving false.

#### Release Notes

* ntheory
  * Fixed outputs for floating point number for isprime

### CODE REVIEW

**Comment by shishir-11** [comment]
```python
def _PrimePredicate_number(expr, assumptions):
    # helper method
    exact = not expr.atoms(Float)
    try:
        i = int(expr.round())
        if (expr - i).equals(0) is False:
            raise TypeError
    except TypeError:
        return False
    if exact:
        return isprime(i)
    # when not exact, we won't give a True or False
    # since the number represents an approximate value
``` 

Any suggestions about the last comment , as discussed in the issue , should we remove and just give false for float values .

File: sympy/ntheory/tests/test_primetest.py [hunk 1]
```diff
@@ -209,7 +209,9 @@ def test_isprime():
     sieve.extend(3000)
     assert isprime(2819)
     assert not isprime(2931)
-    assert not isprime(2.0)
+    # assert not isprime(2.0)
+    raises(ValueError,lambda: isprime(2.0))
+    raises(ValueError,lambda: isprime(Float(34561)))
```
Comment (smichr) [inline sympy/ntheory/tests/test_primetest.py:214]:
```suggestion
    raises(ValueError,lambda: isprime(2.0))
    raises(ValueError,lambda: isprime(Float(2)))
```

File: sympy/ntheory/tests/test_primetest.py [hunk 2]
```diff
@@ -9,7 +9,7 @@
                                      is_mersenne_prime)
 
 from sympy.testing.pytest import slow, raises
-from sympy.core.numbers import I
+from sympy.core.numbers import I,Float
```
Comment (smichr) [inline sympy/ntheory/tests/test_primetest.py:12]:
```suggestion
from sympy.core.numbers import I, Float
```

**Comment by smichr** [comment]
> just give false for float values

If we aren't going to raise an error with `ask` then I think None would be most appropriate since it will cause the user to reflect on why an answer couldn't be given. But I would make it return False if `sympy.numbers.int_valued(number) == False`

**Comment by shishir-11** [comment]
@smichr I tried out a few examples and believe the function presently does what you mentioned . 

```python
>>> ask(Q.prime(2.3))
False
>>> ask(Q.prime(Float(2.3)))
False
>>> ask(Q.prime(Float(2)))
>>> ask(Q.prime(2))
True
>>> ask(Q.prime(2.0))
``` 
Please correct me if there was a mistake in my understanding. Otherwise , I think the current changes should fix the issue.

**Comment by asmeurer** [comment]
None from ask sounds good. It definitely shouldn't raise. That would break the whole design of ask.

**Comment by shishir-11** [comment]
I have tried to fix the problems causing tests to fail and squashed the commits, kindly tell me if anything more is needed. Thanks.

**Comment by smichr** [comment]
Looks good, thanks. It's in.

**Pull request merged**